sale_section_and_note_revamp/models/display_line_mixin.py

```diff
@@ -87,6 +87,7 @@ def _get_section_lines(self, without_notes=True):
             if next_record.is_section():
                 break
             if not without_notes and next_record.is_note():
+                next_record = next_record.next_line_id
                 continue
             result |= next_record
```

Comment:
```suggestion
            if without_notes  or not next_record.is_note():
                result |= next_record
```
I find it a bit more elegant (and it avoids duplicating a line)

Reply:
and actually I'm wondering if the original condition is correct. 

From a logical point of view I want to include the next line 
* if without_note is False (meaning with notes, meaning we want all lines)
* or if it is not a note




```suggestion
            if not without_notes or not next_record.is_note():
                result |= next_record
```

Reply:
It's not a functional correct fix

Reply:
Issue comes from the construction of a cyclic graph in the first place. Instead of a tree one

Reply:
A note is twice refered in two different lines.

Reply:
ok if you say so. I think I'd like a unit test with the fix, in any case :)

Reply:
Fixed in the main branch. Test incoming once customer is delivered.

Reply:
Resolved like this :
`
    def _get_section_lines(self, with_notes=True):
        """Get all order lines in a section.
        :param with_notes: Include note lines
        """
        self.ensure_one()
        if self.is_section():
            result = self.env[self._name]
            next_record = self.next_line_id
            while next_record:
                if next_record.is_section():
                    break
                result |= next_record
                next_record = next_record.next_line_id
            notes = result.filtered(lambda r: r.is_note())
            return result if with_notes else result - notes
`

Reply:
1. I'm surprised to see the method is returning None if called on a line which is not a section
2. I would not make the methods in this mixin exposed to RPC Do you really need to call them through the web client?
3. consider using a common prefix for mixin methods, this will help avoiding collisions when the mixin is applied.